### ISSUE TITLE
Show apps with pending notifications

### ISSUE THREAD

**Issue by enriquedfa** [8bitPit/Niagara-Issues#152]
I just thought that instead of showing the most used apps below favorites, you could add an option to show apps with a pending notification.

For example, someone messaged me on Facebook Messenger, so the Messenger icon appears below favorites, and if I dismiss or answer the notification, messenger would also dissapear..

**Comment by 8bitPit** [comment]
Perhaps, I'll add an option for that

**Comment by jakcharvat** [comment]
Refer to #9

**Comment by 8bitPit** [comment]
There should be a filter though (only for messenger), because too many notifications could clutter up your screen

**Comment by Toontje** [comment]
That would reduce my favorites list to 0 apps and the launcher perfect for me. My preference would be to only show apps with a notification and being able to preview them. No more icons on the home screen.

**Comment by UESPAsputnik** [comment]
I would also much prefer apps with pending notifications instead of app being added (seemingly) randomly to the home screen. That's would make the launcher's homescreen perfect for me. :)

**Comment by praveenchandranpk** [comment]
This will improve the productivity of the launcher very much in my opinion and the homescreen will be much more alive.

**Comment by 8bitPit** [comment]
I want to add that I wouldn't show every notification above your favorites, only the most important ones. For example, "App X was updated", "Screenshot saved", "10% off, only today!" or "Download completed" isn't worth the notification slot on the homescreen imo. I'd say it's better to display only notifications from messages that were sent by (most of the time) real people, like those from messengers, emails, or from private social media chats?

**Comment by tkarika** [comment]
> I want to add that I wouldn't show every notification above your favorites, only the most important ones.

Do you have access to the notification level? Maybe there can be a setting to show only the unmuted, or the muted but not small, or all (if somebody wants to spam their homescreen with a lot of apps).

**Comment by 8bitPit** [comment]
@tkarika yeah, something similar would be possible :)

**Comment by luxcalor** [comment]
Would be really nice to have notifications  for apps in "app-folders" and be able to expand them somehow. Maybe swipe on the main app and then swipe again on the nested. But at least be able to see which apps in the "folder" have notifications.

**Comment by 8bitPit** [comment]
@luxcalor Agreed: #1201

**Comment by berdzena** [comment]
> I want to add that I wouldn't show every notification above your favorites, only the most important ones. For example, "App X was updated", "Screenshot saved", "10% off, only today!" or "Download completed" isn't worth the notification slot on the home screen imo. I'd say it's better to display only notifications from messages that were sent by (most of the time) real people, like those from messengers, emails, or from private social media chats.

It would be great....  to see chat notifications (those that are not added in favorites) on top of favorites..

**Comment by soliddave** [comment]
I would point out that if the status bar is marked hidden, the notification shade is irrelevant without swiping down. That was my reasoning for wanting the alert notifications in the favorites.

**Comment by luxcalor** [comment]
Well, the realization of notifications inside pop-up folders in v1.2 update is flawless. The only thing missing now is ability to see which folder has notifications from which app at a glance on the home screen. Currently I have to swipe each folder to check if there are some new notifications or open the dreaded notification shade. Hope this gets implemented in some way, so we can let the shade be and interact with notifications solely from the home screen. :)

**Comment by PemikM** [comment]
Maybe the solution to prevent clutter from notifications would be to show them in a special pop up, which would appear only if there were any notifications from apps outside the favourites list.

**Comment by coffinshaker** [comment]
First of all sorry for the duplicate (#1649) but I somehow missed this topic.

I tested the beta version with the new Notifications feature. It's nice but it is just the opposite of a feature I would expect to have.

Still there would be nice to have an app to pop-up a notification just below the favourites as soon as it comes and disappear as soon as I dismiss it. Let's say there's a courrier app that informs me about a package ready to pick up. The only way as of now to see this information is either by scrolling through the list or dragging down the notifications area. I'd prefer this app to stick to the main screen until I dismiss the notification. Let's say I go, grab the package and I'm not interested in seeing this app until the next package comes someday.

**Comment by deathau** [comment]
I've only recently started using Niagara launcher, and this feature immediately struck me as something that would be useful. Yes it kinda makes the notification shade redundant (except while in other apps) but I think it might also help with notification overload.

For example, one of the first things I did was create a games folder, and now all my notifications from games (usually junk) are limited to that one line of the folder. I can disable these notifications, but sometimes I want to be reminded that my lives have been refilled or whatever.

So my thoughts were that I could group apps with similar purposes (e.g. work) and notifications from those apps are restricted to their little area to look at when I am in the right headspace, but others will appear in my list as they come in, naturally drawing my attention to the most important notifications, while containing the rest.

**Comment by gustavoholst** [comment]
> That would reduce my favorites list to 0 apps and the launcher perfect for me. My preference would be to only show apps with a notification and being able to preview them. No more icons on the home screen.

Yes I would love this.  A very short list of favorites and then all the stuff that wants my attention. Would love to be able to ignore the notification shade and just use this!

A blacklist (similar to the often used app blacklist) would limit the clutter just enough to make this perfect.

**Comment by McFaul4** [comment]
I think the Notification drawer/quick settings is nowadays almost useless. (With Niagara displaying the notifications) 

There is only the problem with the notifications from these non-favorite apps. I rarely use the quick settings to toggle things _(I have wifi, bt, gps, data always on and power saving, dark mode, eye-comfort is toggled by time/Sunset)_ 
and just use them to access stuff _(like Wifi-Menu, BT-Menu, calculator, vpn-menu)_, which the quick settings simply aren't made for.

That's why it would be very nice if you didn't have to call up this menu anymore. And then you could assign the swipe down gesture to something other.(I experimented with this in Nova) But I have to say that I'm a big fan of rethinking the basic features of Android (also why I love Niagara) and I think a lot of them are the way there are because they always have been, although it's not the best.So most of this probably won't come to Niagara in years but I welcome every step in this direction and i hope it was inspiring. Ty <3

**Comment by jdtemple** [comment]
I love this idea, and just created a duplicate of it that was closed.

I'd like to have any non-favorite apps bubble up to the home screen if there are active notifications on them, for sure.

That means I don't have to go hunting for them in the notification tray or by scrolling through the app list any more, they're just there, either at the top of the favorites, or at the bottom, either way.

Then I can act on them right from my home screen!

**Comment by lucasfais** [comment]
This would also be useful when you have "ongoing" activities like requesting a Uber or food delivery. Being able to preview the last (non-favorite) notification right on the home screen and dismiss them without pulling the notification panel would be great.

**Comment by bean5** [comment]
> Would be really nice to have notifications  for apps in "app-folders" and be able to expand them somehow. Maybe swipe on the main app and then swipe again on the nested. But at least be able to see which apps in the "folder" have notifications. 

I would call this feature "group notifications by folder group". It can work intuitively until an app is in more than 1 folder group.

**Comment by bean5** [comment]
Some of the requests here feel like a customizable widget. That's a good place to place an advanced feature without cluttering the settings menu with words and notions that might confuse out of the context of a widget.

**Comment by bean5** [comment]
> This would also be useful when you have "ongoing" activities like requesting a Uber or food delivery. Being able to preview the last (non-favorite) notification right on the home screen and dismiss them without pulling the notification panel would be great.

If I wrote this, as an engineer, I'd call it "Temporarily give preference to notifications to apps that were recently opened." Programmatically I would consider recently opened apps as temporary favorites until a timer expires.

**Comment by ChristopheRoyer21** [comment]
Some updates about that? I've just started placing some apps insidethe popup of another app and now I'm not getting notifications from the,. I guess this is related to the ticket here somehow
